### PR TITLE
Catch exceptions in greenlight's main function and exit after

### PR DESCRIPTION
**Pull request by DCardenasAmp**
We've seen some instances where an exception in the test runner can cause the JVM to hang indefinitely, such as when the test config is invalid but the test system has some threads still running.

Let's try wrapping our test runner with a try/catch, and logging any exceptions that come up and explicitly exit afterward.

Tested this by throwing an exception in main while having a thread running in the background, and verified that I could reproduce the JVM hanging and that this PR addresses that issue.

#nsq i noticed this project isn't really using `clojure.tools.logging` at all, is that a hard rule? it's technically already being pulled in as a dependency from `amperity/envoy`

### CODE REVIEW

**Review by scgerkin — APPROVED** [review]
makes sense to me

**Review by bencharas — APPROVED** [review]
lgtm

File: src/greenlight/runner.clj [hunk 1]
```diff
@@ -294,7 +295,14 @@
...
           "info" (print-test-info tests options (rest arguments))
-          "test" (run-tests! new-system tests options (rest arguments))
+          "test" (try
+                   (run-tests! new-system tests options (rest arguments))
+                   (catch Exception ex
+                     (println "Uncaught exception in test runner:")
+                     (stacktrace/print-stack-trace ex)
+                     (println "Caused by:")
+                     (stacktrace/print-cause-trace ex)
+                     (*exit* 1)))
```
[the remark this answers is not in the document]
Comment (DCardenasAmp) [reply src/greenlight/runner.clj:305]: ah i must misinterpreted the difference here, thanks!

**Review by brandonvin — APPROVED** [review]
👍

**Pull request merged**